File: CustomerAssignment.Customers.Application/Requests/UpdateCustomerAddressRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace CustomerAssignment.Customers.Application.Requests
{
    public class UpdateCustomerAddressRequest
    {
        [Required]
        public Guid CustomerId { get; set; }
        [Required]
        public string Country { get; set; }
        [Required]
        public string City { get; set; }
        [Required]
        public string Street { get; set; }
        public string HouseNumber { get; set; }
        public string AppartmentNumber { get; set; }
        [Required]
        public string ZipCode { get; set; }
    }
}

File: CustomerAssignment.Clients.Website/wwwroot/app/customers/requests/updateCustomerAddressRequest.ts
namespace CustomerAssignment.Customers.Request {

    export class UpdateCustomerAddressRequest {
        public customerId: string;
        public country: string;
        public city: string;
        public street: string;
        public houseNumber: string;
        public appartmentNumber: string;
        public zipCode: string;
    }
}